plugins/env/lib/samson_env/samson_plugin.rb

```diff
@@ -36,6 +36,7 @@ def write_dotenv(base_file, groups)
       groups.each do |suffix, data|
         generated_file = "#{base_file}#{suffix}"
         File.write(generated_file, generate_dotenv(data))
+        File.chmod(0600, generated_file)
```

Comment:
this would leave the file readable for a short time, prefer using atomic: `File.write generated_file, generate_dotenv(data), 0, perm: 0600`